cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -988,7 +988,7 @@ private predicate indirectExprNodeShouldBeIndirectOperand(RawIndirectOperand nod
       e = e0.getFullyConverted()
     )
     or
-    not indirectExprNodeShouldBeIndirectOperand0(_, _, e.getUnconverted()) and
+    not indirectExprNodeShouldBeIndirectOperand0(_, node, _) and
```

Comment:
The commit says "Fix a bug in Expr <-> Node mapping.". What is the bug exactly?

Reply:
I've gone back and forth to figure out exactly why this fix actually fixes all the missing results. The most high-level explanation (which is the one I currently understand myself 🤔) is that the dataflow node that receives the flow was not the one that was exposed as having a result for `asIndirectExpr()`. For example, consider this piece of code:
```cpp
void sink(char*);
void source(char*);

void test() {
  char buffer[256];
  source(buffer); // *buffer is tainted
  sink(buffer); // will have a array-to-pointer conversion
}
```
Dataflow will step from the output of `source` to the fully-converted `buffer` on the `sink(buffer)` line, but `asIndirectExpr()` would give back the _unconverted_ `buffer`.

Instead, this commit should hopefully ensure that only the fully-converted node is the node that has a result for `asIndirectExpr()`.